Do not write payload if page count is 0

`page_count` is used by `xcp-rrdd` to define the size
to allocate to a `cstruct` to write the payload into.

If `page_count=0` the `cstruct` has a size of 0 and
the attempt of writing the payload results in a
`not enough memory` error

Upstream PR: https://github.com/xapi-project/xen-api/pull/4982

diff --git i/lib/reporter_local.ml w/lib/reporter_local.ml
index 21cb93c..88f83a0 100644
--- i/lib/reporter_local.ml
+++ w/lib/reporter_local.ml
@@ -42,11 +42,13 @@ let start_local (module D : Debug.DEBUG)
           ~uid
           ~protocol
           ~overdue_count:!overdue_count;
-      let payload = Rrd_protocol.({
-          timestamp = Utils.now ();
-          datasources = dss_f ();
-        }) in
-      writer.Rrd_writer_functor.write_payload payload;
+      if page_count > 0 then
+        let payload =
+          Rrd_protocol.{timestamp= Utils.now (); datasources= dss_f ()}
+        in
+        writer.Rrd_writer_functor.write_payload payload
+      else
+        D.debug "page_count is 0: skipping payload write" ;
       Thread.delay 0.003
     in
     let cleanup () =
